API-GUIDELINES.md

```diff
@@ -0,0 +1,54 @@
+# ESP-RS API Guidelines
+
+## About
+
+This is a living document - make sure to check the latest version of this document.
+
+> [!NOTE]
+> Not all of the currently existing code follows this guideline, yet.
+
+In general, the [Rust API Guidelines](https://rust-lang.github.io/api-guidelines) apply to all projects in the ESP-RS GitHub organization where possible. (e.g. `C-SERDE` does not apply) 
```

Comment:
should we list all the rules which don't apply?

Reply:
I think so, if its just a handful